interface/forms/annotate_diagram/mapdiagram/AbstractAnnotateModel.php

```diff
@@ -0,0 +1,282 @@
+<?php
+/**
+ * Copyright Medical Information Integration,LLC info@mi-squared.com
+ * 
+ * LICENSE: This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://opensource.org/licenses/gpl-license.php>;.
+ *
+ * Rewrite and modifications by sjpadgett@gmail.com Padgetts Consulting 2016.
+ *
+ * @file AbstractAnnotateModel.php
+ *
+ * @brief This file contains the AbstractAnnotateModel class, used to model form contents.
+ *
+ * @package OpenEMR
+ * @author  Medical Information Integration,LLC <info@mi-squared.com>
+ * @author  Jerry Padgett <sjpadgett@gmail.com> 
+ * @author  Terry Hill <terry@lillysystems.com>
+ * @link    http://www.open-emr.org
+ */
+
```

Comment:
Use the security model:
http://www.open-emr.org/wiki/index.php/Codebase_Security#SQL-Injection_and_Cross-Scripting_Prevention

$sanitize_all_escapes=true;
$fake_register_globals=false;